pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -318,48 +318,80 @@ func (s *Service) ReconcileLoadBalancerMember(openStackCluster *infrav1.OpenStac
 	return nil
 }
 
+// DeleteLoadBalancer deletes a load balancer identified by its name and all child objects (cascade).
 func (s *Service) DeleteLoadBalancer(openStackCluster *infrav1.OpenStackCluster, clusterName string) error {
 	loadBalancerName := getLoadBalancerName(clusterName)
 	lb, err := s.checkIfLbExists(loadBalancerName)
 	if err != nil {
 		return err
 	}
+	return s.deleteLoadBalancer(openStackCluster, lb)
+}
 
+// deleteLoadBalancer deletes the load balancer and all child objects (cascade).
+func (s *Service) deleteLoadBalancer(openStackCluster *infrav1.OpenStackCluster, lb *loadbalancers.LoadBalancer) error {
 	if lb == nil {
 		return nil
 	}
 
 	if lb.VipPortID != "" {
 		fip, err := s.networkingService.GetFloatingIPByPortID(lb.VipPortID)
 		if err != nil {
-			return err
+			return errors.Wrapf(err, "get floating IP by port ID %s", lb.VipPortID)
 		}
 
 		if fip != nil && fip.FloatingIP != "" {
 			if err = s.networkingService.DisassociateFloatingIP(openStackCluster, fip.FloatingIP); err != nil {
-				return err
+				return errors.Wrapf(err, "dissassociate floating IP %s", fip.FloatingIP)
 			}
 			if err = s.networkingService.DeleteFloatingIP(openStackCluster, fip.FloatingIP); err != nil {
-				return err
+				return errors.Wrapf(err, "delete floating IP %s", fip.FloatingIP)
 			}
 		}
 	}
 
 	deleteOpts := loadbalancers.DeleteOpts{
 		Cascade: true,
 	}
-	s.logger.Info("Deleting load balancer", "name", loadBalancerName, "cascade", deleteOpts.Cascade)
+	s.logger.Info("Deleting load balancer", "id", lb.ID, "name", lb.Name, "cascade", deleteOpts.Cascade)
 	mc := metrics.NewMetricPrometheusContext("loadbalancer", "delete")
-	err = loadbalancers.Delete(s.loadbalancerClient, lb.ID, deleteOpts).ExtractErr()
+	err := loadbalancers.Delete(s.loadbalancerClient, lb.ID, deleteOpts).ExtractErr()
 	if mc.ObserveRequest(err) != nil {
 		record.Warnf(openStackCluster, "FailedDeleteLoadBalancer", "Failed to delete load balancer %s with id %s: %v", lb.Name, lb.ID, err)
-		return err
+		return errors.Wrapf(err, "delete loadbalancer by ID %s", lb.ID)
 	}
 
 	record.Eventf(openStackCluster, "SuccessfulDeleteLoadBalancer", "Deleted load balancer %s with id %s", lb.Name, lb.ID)
 	return nil
 }
 
+// DeleteLoadBalancersByNetworkID deletes all remaining load balancers in the network
+// identified by openStackCluster.Status.Network.ID.
+func (s *Service) DeleteLoadBalancersByNetworkID(openStackCluster *infrav1.OpenStackCluster) error {
```

Comment:
one question here about the logic
I think multiple cluster share one network is a valid use case? e.g I give 10.0.0.16/24 and I want my clusters let's say 2 clusters in this network range , then we will delete all LB in case we delete one cluster? 
should we add some annotation or tag to mark this LB belong to this cluster and only handle update to it?

or we already have `project` so only visible to the LB that belongs to this cluster?

Reply:
The deletion of the load balancer is only executed if `openStackCluster.Spec.NodeCIDR` is set. It looks like the network is anyway deleted some code lines below, so deleting the attached load balancers should be fine.
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/ed56be16af090c926a610ab03a70d6ba96962573/controllers/openstackcluster_controller.go#L158-L169

Maybe I'm missing something, but having 2 clusters in the same network would require at least one cluster that has an empty `openStackCluster.Spec.NodeCIDR` but some values for `openStackCluster.Spec.Network` and `openStackCluster.Spec.Subnet`.

Reply:
>>>the network is anyway deleted some code lines below, so deleting the attached load balancers should be fine.

>>>Maybe I'm missing something, but having 2 clusters in the same network would require at least one cluster that has an empty openStackCluster.Spec.NodeCIDR but some values for openStackCluster.Spec.Network and openStackCluster.Spec.Subnet.

not sure whether there are use case like this , e.g 
2 clusters ,first create with NodeCIDR then 2nd use the Network/Subnet etc

so maybe we can document some restriction(not sure we already have? )
e.g if you want to create multiple clusters, the network/subnet lifecycle is not included in CAPO?
and the network/subnet that create through NodeCIDR automatically should not be used for 
further cluster provisioning (otherwise as you mentioned, it might lead to some issue?)

Reply:
I would expect this to be a use case for multiple networks. i.e. If you want private networking between multiple clusters I would still expect each cluster to have its own control plane network, but share one or more additional dataplane networks.

Have I misunderstood the use case here?